lib/topological_inventory/orchestrator/api.rb

```diff
@@ -121,8 +124,18 @@ def tenant_header(tenant_account)
         {"x-rh-identity" => Base64.strict_encode64({"identity" => {"account_number" => tenant_account}}.to_json)}
       end
 
+      def supported_application_type_ids(timeout = 300)
+        @@cache_lock.synchronize do
+          if @@cache_start.nil? || @@cache_start <= timeout.seconds.ago
+            @@supported_application_type_ids = load_supported_application_type_ids
+            @@cache_start = Time.now
+          end
+        end
+        @@supported_application_type_ids
+      end
```

Comment:
If we were going to do this at the class level it was so that we could use `cache_with_timeout`, if we're going to reimplement the syncronize I don't think it has to be a class level cache but we can reevaluate in a follow-up

Reply:
we need to save it at class level because each sync round creates it's own API object, but we need to use API calls, which are implemented at object level, which doesn't have equivalent for `cache_with_timeout`